Sources/Datadog/DatadogCore/Context/DatadogContextProvider.swift

```diff
@@ -71,11 +80,17 @@ internal final class DatadogContextProvider {
     ///
     /// **Warning:** Must be called from the `queue`.
     private func unsafeRead() -> DatadogContext {
-        var context = self.context
-        reader.read(to: &context)
+        reader.read(to: &self.context)
         return context
     }
 
+    /// Publishes context changes to tge given receiver.
+    ///
+    /// - Parameter receiver: The receiver closure.
+    func publish(to receiver: @escaping ContextValueReceiver<DatadogContext>) {
+        queue.async(flags: .barrier) { self.receivers.append(receiver) }
+    }
```

Comment:
It doesn't seem to be tested in [`DatadogContextProviderTests`](https://github.com/DataDog/dd-sdk-ios/blob/maxep/RUMM-2334/tracing-with-rum-context/Tests/DatadogTests/Datadog/DatadogCore/DatadogContextProviderTests.swift).